### PR TITLE
Always use git clean -fdx

### LINKED ISSUE #274: `AMD64 Arch Linux Asan Debug` runs incomplete `git clean` before building
`AMD64 Arch Linux Asan Debug` runs `git clean -f -f -d` iso. `git clean -fdx`.

### PR DESCRIPTION
**Pull request by erlend-aasland**
Ref. comment in #273. Not sure if this fixes #274; I don't know the buildbot process well enough.

### CODE REVIEW

**Comment by erlend-aasland** [comment]
See also #255

File: master/master.cfg [hunk 1]
```diff
@@ -309,7 +309,7 @@ for name, worker, buildfactory, stability in BUILDERS:
 
     pull_request_builders.append(buildername)
 
-    source = GitHub(repourl=git_url, timeout=3600, mode="full", method="clean",)
+    source = GitHub(repourl=git_url, timeout=3600, mode="full", method="fresh",)
```
Comment (vstinner) [inline master/master.cfg:312]: Oh. There are Git(...) and GitHub(...) calls which get also the same parameters.

Maybe we could share parameter with something like:

```
# with super cool comments here!
GIT_PARAMETER= ...

GitHub(repourl=git_url, **GIT_PARAMETERS)
```

Comment (erlend-aasland) [reply master/master.cfg:312]: Yeah, SGTM

Comment (erlend-aasland) [reply master/master.cfg]: Something like 58dc87ec723f4f6f6fb246d1189da8e29791bbd7?

**Comment by vstinner** [comment]
Perfect, thanks!

**Pull request merged**